### PR TITLE
Feat/412 observer fss

### PR DESCRIPTION
**Pull request by ivan-015**

### CODE REVIEW

File: lib/controllers/interpret.dart [hunk 1]
```diff
@@ -77,8 +79,9 @@ class Interpret {
         var tempTree = currentScreen;
         tempTree.name = designPage.name;
 
+        tempTree.data = PBGenerationViewData();
         if (currentScreen.rootNode is InheritedScaffold) {
-          tempTree.tree_type = TREE_TYPE.SCREEN;
+          PBPlatformOrientationLinkerService().addOrientationPlatformInformation(tempTree);
```
Comment (SushiRoll53) [inline lib/controllers/interpret.dart]: I think we remove `tempTree.tree_type = TREE_TYPE.SCREEN;` by mistake
that should still be part of this

**Review by SushiRoll53 — APPROVED** [review]
LGTM

File: lib/generation/generators/value_objects/file_structure_strategy/command_invoker.dart [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import 'package:parabeac_core/generation/generators/value_objects/file_structure_strategy/commands/file_structure_command.dart';
+
+abstract class CommandInvoker {
```
Comment (x64Eddie) [inline lib/generation/generators/value_objects/file_structure_strategy/command_invoker.dart:3]: I would suggest just placing a small description on what we are using this abstract class for :thinking:

**Review by x64Eddie — APPROVED** [review]

File: lib/generation/generators/value_objects/file_structure_strategy/command_invoker.dart (hunk 2 above)
Comment (ivan-015) [reply lib/generation/generators/value_objects/file_structure_strategy/command_invoker.dart:3]: Good observation. Will add descriptions to this and the other abstract classes from this PR.

**Pull request merged**